stdsimd/arch/detect/cache.rs

```diff
@@ -1,6 +1,8 @@
 //! Caches run-time feature detection so that it only needs to be computed
 //! once.
 
+#![allow(dead_code)] // not used on all platforms
+
```

Comment:
I would prefer if we would just disable the `detect` module in the platforms in which it is not used. I can do this in the MIPS PR.